x-pack/plugins/endpoint/public/embeddables/resolver/models/process_event.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ProcessEvent } from '../types';
+
+/**
+ * Returns true if the process's eventType is either 'processCreated' or 'processRan'.
+ * Resolver will only render 'graphable' process events.
+ */
+export function isGraphableProcess(event: ProcessEvent) {
+  return eventType(event) === 'processCreated' || eventType(event) === 'processRan';
+}
+
+/**
+ * Returns a custom event type for a process event based on the event's metadata.
+ */
+export function eventType(event: ProcessEvent) {
+  const {
+    data_buffer: { event_type_full: type, event_subtype_full: subType },
```

Comment:
I think really all you'll need to change to make this work with the data generator is remove this from being nested inside data_buffer https://github.com/elastic/endgame-evrelay/blob/dev/internal/events/mega.go#L105-#L106